### PR TITLE
Enabled Doxygen locally, documented some interfaces.

### PR DESCRIPTION
**Pull request by mvidner**
This is a reincarnation of #62 which was for the SLE11 branch.

### CODE REVIEW

File: url.h [hunk 1]
```diff
@@ -22,7 +22,7 @@ typedef struct url_data_s {
   unsigned unzip:1;
   unsigned label_shown:1;
   unsigned optional:1;
-  char *compressed;		// program name used for compression, if any
+  char *compressed;          /**< program name used for compression, if any */
```
Comment (kobliha) [inline url.h]: This sounds like a boolean

File: file.h [hunk 2]
```diff
@@ -59,17 +59,17 @@ typedef enum {
...
-  kf_ibft = 1 << 8,		/* ibft values (iSCSI BIOS) */
-  kf_cont = 1 << 9,		/* 'content' file */
-  kf_comma = 1 << 10		/* accept commas as option separator (in command line syntax) */
+  kf_cfg = 1 << 0,		/**< /linuxrc.config & /info */
+  kf_cmd = 1 << 1,		/**< /proc/cmdline, after info */
+  kf_cmd_early = 1 << 2,	/**< /proc/cmdline, before info  */
+  kf_yast = 1 << 3,		/**< /etc/yast.inf */
+  kf_dhcp = 1 << 4,		/**< dhcp info file */
+  kf_mem = 1 << 5,		/**< /proc/meminfo */
+  kf_boot = 1 << 6,		/**< things the boot loader used */
```
Comment (kobliha) [inline file.h]: things?

File: global.h [hunk 3]
```diff
@@ -323,240 +323,240 @@ typedef struct ifcfg_s {
...
-  unsigned early_bash:1;	/* start bash on tty8 */
-  unsigned devtmpfs:1;		/* mount devtmpfs */
-  unsigned plymouth:1;		/* start plymouth */
-  unsigned restarting:1;	/* we are preparing for restart */
-  unsigned restarted:1;		/* we have been restarted */
-  unsigned wicked:1;		/* use wicked for network setup */
-  unsigned nanny:1;		/* use wickedd-nanny */
-  unsigned nanny_set:1;		/* nanny setting was changed */
-  unsigned upgrade:1;		/* upgrade or fresh install */
+  unsigned rebootmsg:1;		/**< show reboot message */
```
Comment (kobliha) [inline global.h]: looks like indentation problem, but maybe just here at GitHub

File: global.h [hunk 4]
```diff
@@ -323,240 +323,240 @@ typedef struct ifcfg_s {
...
+  unsigned redraw_menu:1;	/**< we need a better solution for this */
+  unsigned initrd_has_ldso:1;	/**< instsys contains a dynamic linker */
+  unsigned suppress_warnings:1;	/**< show less warning dialogs */
```
Comment (kobliha) [inline global.h]: fewer?

File: global.h [hunk 5]
```diff
@@ -323,240 +323,240 @@ typedef struct ifcfg_s {
...
-    int delay;			/* wait this much after insmod */
-    driver_t *drivers;		/* list of extra drive info */
-    unsigned disks:1;		/* automatically ask for module disks */
-    slist_t *options;		/* potential module parameters */
+    char *dir;				/**< modules directory */
+    char *type_name[MAX_MODULE_TYPES];	/**< module type names */
+    char *more_file[MAX_MODULE_TYPES];	/**< file name of module archive */
+    int disk[MAX_MODULE_TYPES];		/**< number of module disk */
+    module_t *list;			/**< list of all modules */
+    int scsi_type;		/**< for some reasons... */
```
Comment (kobliha) [inline global.h]: great comment :)

File: global.h [hunk 6]
```diff
@@ -323,240 +323,240 @@ typedef struct ifcfg_s {
...
+    int64_t load_image;		/**< _load_ rootimage, if we have at least that much */
+    int64_t ram;		/**< ram size */
+    int64_t ram_min;		/**< min required memory (ram size) needed for install */
   } memoryXXX;
 
-  struct {			/* mountpoints */
-    unsigned cnt;		/* mp counter */
-    unsigned initrd_parts;	/* initrd parts counter */
+  struct {			/**< mountpoints */
+    unsigned cnt;		/**< mp counter */
```
Comment (kobliha) [inline global.h]: mp, mountpoints?

File: util.c [hunk 7]
```diff
@@ -529,12 +529,12 @@ void util_truncate_dir(char *dir)
...
-/*
+/**
  * Check whether 'file' exists and return file type.
  *
- * return:
- *   0: does not exists
- *   'r', 'd', 'b', 1: type (1: other)
+ * @return
+ * -  0: does not exist.
+ * -  'r', 'd', 'b', 1: type (1: other)
```
Comment (kobliha) [inline util.c]: maybe elaborate a bit on these return values?

**Comment by kobliha** [comment]
@mvidner What's the status of this PR? Worth merging?

@wfeldt Except that in needs to be rebased, do you have any comments?

**Comment by wfeldt** [comment]
Although I think that doxygen is not well suited for non-oo languages, I'm not opposed to it.
But the patch has to be rebased.

**Comment by mvidner** [comment]
@wfeldt I have updated this, removing the big change of marking all comments like this
```diff
-  kf_cfg = 1 << 0,		/* /linuxrc.config & /info */
+  kf_cfg = 1 << 0,		/**< /linuxrc.config & /info */
```

**Comment by mvidner** [comment]
And I have added them again

**Comment by wfeldt** [comment]
ok, lets try

**Pull request merged**